File: 算法导论/第二章-算法基础/sort.py
# 算法导论  => 第二章  => 插入排序
import random
import time
print("正在生成随机数组...")
arr = random.sample(range(1, 300000), 10000)
# arr = [5, 3]
print("随机数组生成成功！一共%d个"%(len(arr)))

# 插入排序
def insertSort(list):
  # print(list)
  # print("-------\n\n\n")
  arr = list
  j = 1
  while j < len(arr):
    key = arr[j]
    i = j - 1
    while i >= 0 and arr[i] > key:
      arr[i + 1] = arr[i]
      i = i - 1
    j = j + 1
    arr[i + 1] = key
  # print(arr)

start = time.perf_counter()

insertSort(arr)

end = time.perf_counter()

print("插入排序数组%d个, 耗时%ss"%(len(arr), (end - start)))

File: 算法导论/第二章-算法基础/sort.js
const {
  performance
} = require('perf_hooks');


// 插入算法

const arr = []
const sum = 100000
const total = 300000000

console.log('随机数组生成中....')
for (let i = 0; i < sum; i++) {
  arr.push(~~(Math.random() * total))
}
console.log(`随机数组生成功! 一共${arr.length}位`)

const insertSort = (list) => {
  // console.log(list)
  const arr = list
  for (let j = 1; j < list.length; j++) {
    let key = arr[j]
    let i = j - 1
    while (i >= 0 && arr[i] > key) {
      arr[i + 1] = arr[i]
      i--
    }
    arr[i + 1] = key
  }
  // console.log(arr)
}

start = performance.now();
insertSort(arr)
end = performance.now();
console.log(`插入排序成功, 花费时间${end - start}ms`)